samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -78,38 +77,38 @@
    * Allows sending {@link MessageEnvelope}s in this {@link MessageStream} to an output
    * {@link org.apache.samza.system.SystemStream} using the provided {@link SinkFunction}.
    *
-   * @param sinkFn  the function to send {@link MessageEnvelope}s in this stream to output systems
+   * @param sinkFn the function to send {@link MessageEnvelope}s in this stream to output systems
    */
   void sink(SinkFunction<M> sinkFn);
 
   /**
-   * Groups the {@link MessageEnvelope}s in this {@link MessageStream} according to the provided {@link Window} semantics
+   * Groups and processes the {@link MessageEnvelope}s in this {@link MessageStream} according to the provided {@link Window}
    * (e.g. tumbling, sliding or session windows) and returns the transformed {@link MessageStream} of
-   * {@link WindowOutput}s.
+   * {@link WindowPane}s.
```

Comment:
I've added an ASCII diagram with a description. Hopefully, it should be clearer now.

 * Groups the incoming {@link MessageEnvelope}s in the {@link org.apache.samza.operators.MessageStream} into finite windows for processing.
``` 
 * <p> Each window is uniquely identified by its {@link WindowKey}. A window can have one or more associated {@link Trigger}s
 * that determine when results from the {@link Window} are emitted. Each emitted result contains one or more
 * {@link MessageEnvelope}s in the window and is called a {@link WindowPane}.
 *
 * <p> A window can have early triggers that allow emitting {@link WindowPane}s speculatively before all data for the window
 * has arrived or late triggers that allow handling of late data arrivals.
```